packages/story-editor/src/elements/shape/icon.js

```diff
@@ -39,12 +39,15 @@ const Container = styled.div`
   background-color: ${({ theme }) => theme.colors.opacity.black10};
 `;
 
-const ShapePreview = styled.div`
+const ShapePreview = styled.div.attrs(({ backgroundColor }) => ({
+  // https://stackoverflow.com/a/58095676
+  // Removes console warning where class was re-generated over 200 times.
+  style: generatePatternStyles(backgroundColor),
+}))`
   width: 100%;
   height: 100%;
   margin: 1px;
   clip-path: ${({ maskId }) => `url(#${maskId})`};
-  ${elementWithBackgroundColor}
```

Comment:
This is not the only usage of this variable. Why does only this one need to be changed / why are others not affected?

Reply:
Yeah, mentioned that in the description. We could fix it in all places, but before we do that I think it'd be best to verify that this is the change we want to make.

This happens in a few places, and I'm not sure why it's only happening to `ShapePreview`. I think this points to a worse problem where the `ShapePreview` is being re-rendered over 200 times.

---

My initial hypothesis was that a single test (or multiple) would throw this warning when something re-rendered the `ShapePreview` component a bunch of times. However, I ran the tests locally and added a log to see which test was running when the warning was logged. Running those tests (and suites) individually didn't log the warning. It was also a different test every time.

So with that, my current assumptions and a possible hypothesis:

**Assumptions**
1. `Styled components` is not keeping track of the `ShapePreview` styled component between tests. Should only matter with one test at a time.
2. Each `ShapePreview` in the layers panel has a different ID.
3. Well under 200 elements are not rendered in the layers panel in all of the automated tests.

**New Hypothesis**
Old test suites aren't being torn down. The `ShapePreview` re-renders in the layer panel multiple times while it remains alive.

Reply:
💭 Unsure yet of where styled-components is adding the class to the document, but If the document styles aren’t cleared, then styled components will keep adding a new class every time the editor is rendered (as well as a new layer)

May be that after 200 layers are added, styled components can tell that they all have very similar ids + styles but they're all different class definitions, so it logs that error

Reply:
Update: verified that after the `ShapePreview` component has been rendered in the test suite 200 times, this error gets logged.

---


1. Every time a new test is started in the editor, a layer is is rendered in the layer panel with the `ShapePreview` element. This will create a `ShapePreview` element.
    - Note: this may not be 200 tests because some tests render more than one layer with a `ShapePreview`
2. Styled components keeps track of the amount of classes that are generated from each component. However, that cache ([seems like it's a cache](https://github.com/styled-components/styled-components/blob/30dab74acedfd26d227eebccdcd18c92a1b3bd9b/packages/styled-components/src/utils/createWarnTooManyClasses.ts#L12)) is not reset between tests.
3. After 200 layers with shape elements have been rendered, the warning gets logged to the console.

This happened only for shape elements because the background layer is a shape and that number gets to 200 quicker than text layers do.

Reply:
Nice.